src/mod_muc_db_rdbms.erl

```diff
@@ -25,122 +25,134 @@
 %% User's JID. Can be on another domain accessable over FED.
 %% Only bare part (user@host) is important.
 -type client_jid() :: jid:jid().
-
+-type room_id() :: pos_integer().
 -type room_opts() :: [{OptionName :: atom(), OptionValue :: term()}].
-
 -type aff() :: atom().
+-type updated_result() :: {updated, non_neg_integer()}.
 
 
 -spec init(server_host(), ModuleOpts :: list()) -> ok.
 init(ServerHost, _Opts) ->
+    prepare_queries(ServerHost),
+    ok.
+
+prepare_queries(ServerHost) ->
+    %% Queries to muc_rooms table
+    prepare(muc_insert_room, muc_rooms,
+            [muc_host, room_name, options],
+            <<"INSERT INTO muc_rooms (muc_host, room_name, options)"
+              " VALUES (?, ?, ?)">>),
+    prepare(muc_select_room_id, muc_rooms,
+            [muc_host, room_name],
+            <<"SELECT id FROM muc_rooms "
+              "WHERE muc_host = ? AND room_name = ?">>),
+    prepare(muc_select_room, muc_rooms,
+            [muc_host, room_name],
+            <<"SELECT id, options FROM muc_rooms "
+              "WHERE muc_host = ? AND room_name = ?">>),
+    prepare(muc_delete_room, muc_rooms,
+            [muc_host, room_name],
+            <<"DELETE FROM muc_rooms WHERE muc_host = ? AND room_name = ?">>),
+    prepare(muc_select_rooms, muc_rooms, [muc_host],
+            <<"SELECT id, room_name, options FROM muc_rooms WHERE muc_host = ?">>),
+    %% Queries to muc_room_aff table
+    prepare(muc_insert_aff, muc_room_aff,
+            [room_id, luser, lserver, resource, aff],
+            <<"INSERT INTO muc_room_aff"
+              " (room_id, luser, lserver, resource, aff)"
+              " VALUES(?, ?, ?, ?, ?)">>),
+    prepare(muc_select_aff, muc_room_aff,
+            [room_id],
+            <<"SELECT luser, lserver, resource, aff "
+              "FROM muc_room_aff WHERE room_id = ?">>),
+    prepare(muc_delete_aff, muc_room_aff, [room_id],
+            <<"DELETE FROM muc_room_aff WHERE room_id = ?">>),
+    %% Queries to muc_registered table
+    prepare(muc_select_nick_user, muc_registered,
+            [muc_host, lserver, nick],
+            <<"SELECT luser FROM muc_registered WHERE muc_host = ?"
```

Comment:
does it mean that we can have one and the same nick reused by the users from the different hosts?

Reply:
I think it's checking for conflicts inside one subdomain.
If we don't check for muc_host, than if someone hosts two domains: `muc.cats.com` and `muc.cars.com`, than there could not be duplicates between these two. Which is something we probably don't wanna do, because users on `cats` could not know anything about `cars`.